app/src/main/java/com/simply/birthdayapp/presentation/ui/components/CleanableSearchBar.kt

```diff
@@ -0,0 +1,100 @@
+package com.simply.birthdayapp.presentation.ui.components
+
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.foundation.text.selection.LocalTextSelectionColors
+import androidx.compose.foundation.text.selection.TextSelectionColors
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Clear
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.simply.birthdayapp.R
+
+@Composable
+fun CleanableSearchBar(
+    query: String,
+    onQueryChange: (String) -> Unit = {},
+    onSearch: () -> Unit = {},
+) {
+    CompositionLocalProvider(
```

Comment:
Do we need a CompositionLocalProvider here to provide TextSelectionColors? We can assign these colors in TextFieldDefaults.colors too

Reply:
[Replaced](https://github.com/SimplyTechnologies/android-internship/pull/14/commits/b48476762ff93f202c2c9f23dee7d17ac108998e)